libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -343,6 +374,19 @@ bool AP_ICEngine::throttle_override(uint8_t &percentage)
         percentage = (uint8_t)start_percent.get();
         return true;
     }
+
+    if (state == ICE_RUNNING && redline_flag && options & !(uint16_t(Options::DISABLE_REDLINE_GOVERNOR))) {
+        // limit the throttle from increasing above what the current output is
+        percentage = SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
+        if (rpm.get_rpm(rpm_instance-1, rpm_value)) {
+            if (filtered_rpm_value > redline_rpm) {
+                // reduce the throttle if still over the redline RPM
+                static const float redline_setpoint_step = idle_slew * AP::scheduler().get_loop_period_s();
+                percentage = SRV_Channels::get_output_scaled(SRV_Channel::k_throttle) - redline_setpoint_step;
```

Comment:
I think you will have to increase the step here, this will only have drop by one step. The change to the scaled output will be cleared after the loop.

Reply:
I had an integrator memory variable in there earlier and then I started thinking I could just use the servo output as that memory. Should have stuck with looking at what the idle governor does. Fixed and reworked the top end throttle limit because of the loop reset. It does get reset to the input each loop before it goes through the overrides/other things that change it.